File: dataset/human/java/file_98.java
import java.math.BigInteger;
import java.util.Arrays;
import java.util.Scanner;
import java.util.stream.IntStream;

public class Main {
    public static void main(String[] args) {
        Scanner std = new Scanner(System.in);
        int N = std.nextInt();
        int[] A = IntStream.range(0, N).map(it -> std.nextInt()).toArray();
        long minusCount = Arrays.stream(A).filter(it -> it < 0).count();
//        long BMax = 0;
        BigInteger BMax = BigInteger.valueOf(0L);
        if (minusCount % 2 == 0) {
            int[] s = Arrays.stream(A).map(Math::abs).toArray();
            for (int i = 0; i < s.length; i++) {
                BMax = BMax.add(BigInteger.valueOf(s[i]));
            }
        } else {
            int[] arrays = Arrays.stream(A).map(Math::abs).sorted().toArray();
            int min = arrays[0];
            for (int i = 1; i < arrays.length; i++) {
                BMax = BMax.add(BigInteger.valueOf(arrays[i]));
            }

            BMax = BMax.subtract(BigInteger.valueOf(min));
        }

        System.out.println(BMax.toString());
    }
}

File: dataset/human/c/file_98.c
#include<stdio.h>


int main(void){
    int count=0;
    int n=0;
    int R=0,G=0,B=0,N=0;
    scanf("%d %d %d %d",&R,&G,&B,&N);
    for(int i=0;i<=N;i++){
        if(i*R>N) break;
        for(int j=0;j<=N;j++){
            if(i*R+j*G>N) break;
            n=N-(i*R+j*G);
            if(n%B==0){
                count++;
            }
        }
    }
    printf("%d\n",count);  
    return 0;
}
